File: dataStructures/LinkedList/SinglyLinkedList/SimpleImplementationAndPrint.java
package dataStructures.LinkedList.SinglyLinkedList;

/**
 * SimpleImplementationAndPrint
 */
public class SimpleImplementationAndPrint {
  Node head;

  static class Node {
    int data;
    Node next;

    Node(int data)
    {
      this.data = data;
      this.next = null;
    }
  }

  public void printLinkedList()  {
    Node n = head;
    while (n != null)
    {
      System.out.print(n.data + " ");
      n = n.next;
    }
    System.out.println();
  }

  public static void main(String[] args) {
    SimpleImplementationAndPrint linkedList = new SimpleImplementationAndPrint();
    linkedList.head = new Node(1);
    Node second = new Node(2);
    Node third = new Node(3);

    linkedList.head.next = second;
    second.next = third;
    third.next = null;

    linkedList.printLinkedList();
  }
}

File: dataStructures/LinkedList/SinglyLinkedList/simple_implementation_and_print.py
class Node:
  def __init__(self, data):
    self.data = data
    self.next = None

class LinkedList:
  
  def __init__(self):
    self.head = None;


  def print_linked_list(self):
    node = self.head
    while node:
      print(node.data)
      node = node.next


if __name__ == "__main__":
    linked_list = LinkedList()
    head = Node(1)
    second = Node(2)
    third = Node(3)

    linked_list.head = head
    linked_list.head.next = second
    second.next = third
    third.next = None

    linked_list.print_linked_list()